custom_components/localtuya/fan.py

```diff
@@ -32,13 +32,16 @@ def flow_schema(dps):
         vol.Optional(CONF_FAN_SPEED_CONTROL): vol.In(dps),
         vol.Optional(CONF_FAN_OSCILLATING_CONTROL): vol.In(dps),
         vol.Optional(CONF_FAN_SPEED_LOW, default=SPEED_LOW): vol.In(
-            [SPEED_LOW, "1", "2", "small"]
+            [SPEED_LOW, "small",
+             "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"]
```

Comment:
Maybe we can do something like:

```python
[SPEED_LOW, "small"] + [str(x) for x in range(1, 11)]
```
Same for the others.

Reply:
is it possible to make it more dynamic, so that the number of speeds can be anything people want.
My fan has 6, this has 10. how can this be accomplished

Reply:
If the values was only integers, we could use `vol.Range` with decent limits. But since it's a mix of integers and strings, there's not really a better way as I see it. We kinda adjust values and ranges along the way, according to what devices need. We could of course just provide a large set of speeds, let's say 0-1000, but it seems unnecessary as no device probably use that.

Reply:
This is really just a patch until the whole fan.py is written to the new percentage based speed.
That’s why i just did 1-10. 

There is a issue for this that probably needs some discussion on how best to do this. 
[https://github.com/rospogrigio/localtuya/issues/383](url)